Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -828,10 +828,18 @@ class Gx2Fitter {
           } else if (measDim == 2) {
             addToGx2fSums<2>(aMatrix, bVector, chi2sum, jacobianFromStart,
                              trackState, *m_addToSumLogger);
+          } else if (measDim == 3) {
+            addToGx2fSums<3>(aMatrix, bVector, chi2sum, jacobianFromStart,
+                             trackState, *m_addToSumLogger);
+          } else if (measDim == 4) {
+            addToGx2fSums<4>(aMatrix, bVector, chi2sum, jacobianFromStart,
+                             trackState, *m_addToSumLogger);
+          } else if (measDim == 5) {
+            addToGx2fSums<5>(aMatrix, bVector, chi2sum, jacobianFromStart,
+                             trackState, *m_addToSumLogger);
           } else {
-            ACTS_ERROR("Can not process state with measurement with "
-                       << measDim << " dimensions.")
-            countNdf -= measDim;
```

Comment:
I wonder if we should keep the error case and explicitly check against `6` as well. Just to catch invalid input early

Reply:
I also was thinking about that. Since we will almost always have 1-3 measurements, we never come that far, so the cost is negligible.

I was wondering if we should add a real error here? I was even thinking of an exception, since 7+ dim/0- dim measurements are a result from a faulty setup?

Reply:
Anything that jumps into the users face would be good IMO. Could be an assert / exception / error. I think I would go with an error in this case.